### PR TITLE
All: Resolve most jQuery Migrate warnings

### PR DESCRIPTION
**Pull request by mgol**
This also adds a new internal `$.ui.__safeOffset__` method to provide a similar support to past jQuery versions while avoiding Migrate warnings.

### CODE REVIEW

File: ui/widgets/tooltip.js [hunk 1]
```diff
@@ -331,7 +331,7 @@ $.widget( "ui.tooltip", {
 					position( positionOption.of );
 					clearInterval( delayedShow );
 				}
-			}, $.fx.interval );
+			}, 13 );
```
Comment (fnagel) [inline ui/widgets/tooltip.js:334]: Do we want a comment here why its 13?

File: ui/safe-offset.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+		// Browser globals
+		factory( jQuery );
+	}
+} ( function( $ ) {
+return $.ui.__safeOffset__ = function( element ) {
+
+	// Simulate a jQuery short-circuiting when there are no client rects reported
+	// which usually means a disconnected node. This check in jQuery is meant just
+	// for IE but UI depends on it.
```
Comment (fnagel) [inline ui/safe-offset.js:16]: Just for my understanding: this check is implemented in jQuery for IE only, bit UI needs it in every browser, right?

File: ui/widgets/tooltip.js (hunk 1 above)
Comment (mgol) [reply ui/widgets/tooltip.js:334]: Good point. This is copied from the `$.fx.interval` value & follows the advice from https://github.com/jquery/jquery-migrate/blob/master/warnings.md#jqmigrate-jqueryfxinterval-is-deprecated

File: ui/safe-offset.js (hunk 2 above)
Comment (mgol) [reply ui/safe-offset.js:16]: Not exactly. jQuery has a check meant for IE but it's run by all browsers so by pure coincidence it makes UI not hit errors here. This is about these lines:
https://github.com/jquery/jquery/blob/3.5.1/src/offset.js#L98-L104

Comment (mgol) [reply ui/safe-offset.js:16]: How about:
```suggestion
	// for IE but is applied in all browsers & UI depends on it.
```
?

Comment (fnagel) [reply ui/safe-offset.js:16]: I still don't understand why we need this in jQueryUI when its already in jQuery... sorry, my mind is a little slow today ;)

Comment (mgol) [reply ui/safe-offset.js:16]: See https://jquery.com/upgrade-guide/3.0/#breaking-change-invalid-input-to-the-offset-method

jQuery 3.0.0 & newer require DOM elements with the `getBoundingClientRect` method. It's just that jQuery needs to run a check for `getClientRects()` due to the comment above the lines I linked, to workaround an IE issue. We shouldn't rely on this check for other purposes.

Also, jQuery Migrate warns if you use `offset()` in such a way.

That said, now I see the discussion in https://github.com/jquery/jquery/issues/2310 so maybe I need to re-read it & make sure Migrate is not warning too much.

**Comment by mgol** [comment]
@fnagel There are some things to check for me in comments about offset changes. I didn't want to delay other changes so I extracted the second commit to #1922 & only landed the first one here.

File: ui/safe-offset.js (hunk 2 above)
Comment (fnagel) [reply ui/safe-offset.js]: Ahhh, got it. Thanks for the explanation and your patience :-)

Comment (mgol) [reply ui/safe-offset.js]: @fnagel Thank you for raising these concerns! This change will most likely not be needed as we'll update Migrate instead.

**Pull request merged**